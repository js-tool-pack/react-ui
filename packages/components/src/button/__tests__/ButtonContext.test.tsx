import { ButtonContext, ButtonGroup, Button } from '..';
import { render } from '@testing-library/react';

describe('ButtonContext', () => {
  test('size', () => {
    // group 的 size 优先
    expect(
      render(
        <ButtonContext.Provider value={{ size: 'large' }}>
          <ButtonGroup size="small">
            <Button></Button>
            <Button></Button>
          </ButtonGroup>
        </ButtonContext.Provider>,
      ).container.firstChild,
    ).toMatchSnapshot();

    // 组件的 props 优先
    expect(
      render(
        <ButtonContext.Provider value={{ size: 'large' }}>
          <Button></Button>
          <Button size="small"></Button>
        </ButtonContext.Provider>,
      ).container.children,
    ).toMatchSnapshot();
  });
  test('other', () => {
    expect(
      render(
        <ButtonContext.Provider value={{ type: 'primary' }}>
          <Button></Button>
          <Button type="info"></Button>
        </ButtonContext.Provider>,
      ).container.children,
    ).toMatchSnapshot();
  });
});
