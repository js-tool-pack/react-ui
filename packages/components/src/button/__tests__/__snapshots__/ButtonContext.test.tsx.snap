// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonContext other 1`] = `
HTMLCollection [
  <button
    class="t-button t-button--type-primary t-button--icon-l t--size-m"
  />,
  <button
    class="t-button t-button--type-info t-button--icon-l t--size-m"
  />,
]
`;

exports[`ButtonContext size 1`] = `
<div
  class="t-button-group"
  role="group"
>
  <button
    class="t-button t-button--type-default t-button--icon-l t--size-sm"
  />
  <button
    class="t-button t-button--type-default t-button--icon-l t--size-sm"
  />
</div>
`;

exports[`ButtonContext size 2`] = `
HTMLCollection [
  <button
    class="t-button t-button--type-default t-button--icon-l t--size-lg"
  />,
  <button
    class="t-button t-button--type-default t-button--icon-l t--size-sm"
  />,
]
`;
